index.html

```diff
@@ -762,18 +839,137 @@ <h2><dfn>CaptureController</dfn></h2>
+                    Create one internal slot: <dfn data-dfn-for="CaptureController">[[\initialFocusLost]]</dfn>,
+                    initialized to <code>false</code>.
+                    <div class="note">
+                      This internal slot ensures that calls to {{CaptureController/setFocusBehavior()}} fail
+                      without throwing if the user manually focuses another tab or native application.
+                    </div>
+                  </p>
+                </li>
+                <li>
+                  <p>
+                    Create one internal slot: <dfn data-dfn-for="CaptureController">[[\focusWindowClosed]]</dfn>,
+                    initialized to <code>false</code>.
+                    <div class="note">
+                      This internal slot tracks explicit closure of the window of opportunity, which happens when
+                      {{CaptureController/setFocusBehavior()}} is called, or the task to close the window executes.
+                      As applications can easily avoid such errors, {{CaptureController/setFocusBehavior()}} when
+                      this slot is set to <code>true</code> throw.
+                    </div>
+                  </p>
+                </li>
+              </ol>
+            </dd>
+            <dt>
+              <dfn>setFocusBehavior</dfn>
+            </dt>
             <dd>
               <p>
-                Create one internal slot: <dfn data-dfn-for="CaptureController">[[\isBound]]</dfn>,
-                initialized to <code>false</code>.
+                Run the following steps:
               </p>
+              <ol>
+                <li>
+                  <p>
+                    Let <var>focusBehavior</var> be the method's first argument.
+                  </p>
+                </li>
+                <li>
+                  <p>
+                    If {{CaptureController/[[isBound]]}} is <code>false</code>,
+                    [=exception/throw=] an "{{InvalidStateError}}" {{DOMException}}.
+                  </p>
+                </li>
+                <li>
+                  <p>
+                    If {{CaptureController/[[videoTrack]]}} is <code>null</code>,
+                    [=exception/throw=] a "{{InvalidStateError}}" {{DOMException}}.
+                  </p>
+                </li>
+                <li>
+                  <p>
+                    If {{CaptureController/[[videoTrack]]}}.{{MediaStreamTrack/getSettings()}}.{{displaySurface}}
+                    is neither {{DisplayCaptureSurfaceType/"browser"}} nor {{DisplayCaptureSurfaceType/"window"}},
+                    [=exception/throw=] a "{{NotAllowedError}}" {{DOMException}}.
+                  </p>
+                </li>
+                <li>
+                  <p>
+                    <!-- TODO: Fix the ReadyState reference. -->
+                    If {{CaptureController/[[videoTrack]]}}'s
+                    <code>ReadyState</code> is NOT {{MediaStreamTrackState/"live"}},
```

Comment:
How about checking first if videoTrack's ReadyState is NOT "live" before videoTrack.getSettings().displaySurface? In other words, switching "4." and "5.".

Reply:
What's the rationale?

Reply:
https://www.w3.org/TR/mediacapture-streams/#dom-mediastreamtrack-getsettings algorithm already checks if track's ReadyState is "ended". I thought we could spare one check.

Reply:
I don't understand how that saves a step. `getSettings()` checks the readyState so that it may produce the same values from before ending. It does not bail early or return anything different when that happens. Or have I misread it?

Also, I'd not want to build a dependency on the specifics of `getSettings()`. That seems brittle.

Reply:
Note that I don't feel strongly about this.

> Also, I'd not want to build a dependency on the specifics of getSettings(). That seems brittle.

Agree.

Reply:
We're good. Feel free to close/resolve this conversation.

Reply:
Cool.